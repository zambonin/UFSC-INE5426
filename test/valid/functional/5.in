int t[10]
int output
output = fold(lambda int x, y -> x + y, t)
