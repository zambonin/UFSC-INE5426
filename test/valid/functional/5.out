int array: t (size: 10)
int var: output
int fun: t_fold (params: int array t)
  int fun: lambda (params: int x, int y)
    ret + x y
  int var: t_tv
  = t_tv [index] t 0
  int var: t_ti
  for: = t_ti 1, < t_ti [len] t, = t_ti + t_ti 1
  do:
    = t_tv + t_tv lambda[2 params] t_tv [index] t t_ti
  ret t_tv
= output t_fold[1 params] t
